Chronos/OneShotDispatchTimer.swift

```diff
@@ -0,0 +1,213 @@
+    deinit { cancel() }
+
+    // MARK: Creating a OneShotDispatchTimer
+
+    /**
+     Creates a OneShotDispatchTimer object.
+
+     - parameter delay:           The execution delay, in seconds.
+     - parameter closure:         The closure to execute after the given delay.
+
+     - returns: A newly created OneShotDispatchTimer object.
+     */
+    convenience public init(delay: Double, closure: OneShotExecutionClosure) {
+        let name = "\(queuePrefix).\(NSUUID().UUIDString)"
+        let queue = dispatch_queue_create((name as NSString).UTF8String, DISPATCH_QUEUE_SERIAL)
+        self.init(delay: delay, closure: closure, queue: queue)
+    }
+
+    /**
+     Creates a OneShotDispatchTimer object.
+
+     - parameter delay:           The execution delay, in seconds.
+     - parameter closure:         The closure to execute after the given delay.
+     - parameter queue:           The queue that should execute the given closure.
+
+     - returns: A newly created OneShotDispatchTimer object.
+     */
+    convenience public init(delay: Double, closure: OneShotExecutionClosure, queue: dispatch_queue_t) {
+        self.init(delay: delay, closure: closure, queue: queue, failureClosure: nil)
+    }
+
+    /**
+     Creates a OneShotDispatchTimer object.
+
+     - parameter delay:           The execution delay, in seconds.
+     - parameter closure:         The closure to execute after the given delay.
+     - parameter queue:           The queue that should execute the given closure.
+     - parameter failureClosure:  The closure to execute if creation fails.
+
+     - returns: A newly created OneShotDispatchTimer object.
+     */
+    public init(delay: Double, closure: OneShotExecutionClosure, queue: dispatch_queue_t, failureClosure: FailureClosure) {
+
+        if let timer = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, queue) {
+            self.timer = timer
+            self.valid = State.valid
+        } else {
+            if let failureClosure = failureClosure {
+                failureClosure()
+            } else {
+                print("Failed to create dispatch source for timer.")
+            }
+        }
+
+        self.queue      = queue
+        self.delay      = delay
+        self.closure    = closure
+
+        super.init()
+
+        weak var weakSelf: OneShotDispatchTimer? = self
+
+        if let timer = timer {
+            dispatch_source_set_event_handler(timer) {
+                if let strongSelf = weakSelf {
+                    strongSelf.cancel() //Ensure timer only gets called once
+                    strongSelf.closure(strongSelf)                    
+                }
+            }
+        }
+    }
+
+    // MARK: Using a OneShotDispatch Timer
+
+    /**
+     Starts the timer.
+
+     - parameter now:     true, if the timer should fire immediately.
+     */
+    public func start(now: Bool) {
```

Comment:
Extending the `Timer` protocol becomes sort of awkward because of the `start` function here, right? For a `OneShotTimer`, the `now` argument doesn't really make much sense. 

I think what we should do is change the declaration of the `start` function in `Timer` from `func start(now: Bool)` to  `func start()`. Then we add `func start(now: Bool)` to the `RepeatingTimer` protocol. What do you think?